lib/index.js

```diff
@@ -1227,6 +1229,10 @@ addMethod('destroy', function () {
 });
 
 Stream.prototype._writeOutgoing = function _writeOutgoing(token) {
+    if (this._explicitly_destroyed) {
+        return;
+    }
+
     if (this._nil_pushed && !this._explicitly_destroyed) {
```

Comment:
Can you remove the check for `!this._explicitly_destroyed` here? Also in `__next_fn`. It's always true now.